pokerbot.js

```diff
@@ -208,4 +256,55 @@ pokerbot.vote = function (req, res, next) {
   return res.status(200).json(responseEphemeral)
 }
 
+/**
+ * We are asking the slack server to give us INformation of all users in channel
+ */
+pokerbot.getAllUsersInTeam = function () {
+  console.log('Inside getAllUsersInTeam : begin')
+  util.getAllUsersInTeam()
+  .then((users) => {
+    for (let index = 0; index < users.length; index++) {
+      pokerbot.allUsersInTeam[users[index].id] = users[index].name
+    }
+    console.log('Got all users in team from  slack.')
+    console.log(pokerbot.allUsersInTeam)
+    console.log('Inside getAllUsersInTeam : end')
+  })
+  .catch((err) => {
+    console.error(err)
+    console.log('Inside getAllUsersInTeam : end')
+  })
+}
+
+/**
+ * We are asking the slack server to give us channel information
+ * @param {Array} userIdArray - Array of all new users.
+ * @param {Function} callback - callback to be executed..
+ */
+pokerbot.addNewUsersInTeam = function (userIdArray, callback) {
+  console.log('Inside addNewUsersInTeam : begin')
+  let functionArray = []
+  let newFunction
+  for (let index = 0; index < userIdArray.length; index++) {
+    let userId = userIdArray[index]
+    newFunction = function () {
+      console.log('Executing util.getUserInTeam with userId : ' + userId)
+      util.getUserInTeam(userId)
+      .then((user) => {
+        console.log('Got user in team from  slack.')
+        console.log(user)
+        pokerbot.allUsersInTeam[user.id] = user.name
+      })
+      .catch((err) => {
+        console.error(err)
+      })
+    }
+    functionArray.push(newFunction)
+  }
```

Comment:
I don't understand why we're creating an array of function calls and then calling them with a new utility function using `async.parallel`. Why can't you just create an array of promises and then do `return Promise.all(promises)` ?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all